datafusion/functions/src/unicode/substr.rs

```diff
@@ -174,18 +174,18 @@ fn string_view_substr(
                     if length == 0 {
                         builder.append_null();
                     } else if length > 12 {
-                        let buffer_index = (*raw >> 64) as u32;
-                        let offset = (*raw >> 96) as u32;
+                        let view = ByteView::from(*raw);
```

Comment:
this is @XiangpengHao's proposal to use ByteView.

However, this code still seems a but strange as it calls `value_unchecked` (which also checks the length again) rather than directly accessing the data in the appropriate buffer. Not sure if we should fix that